kpops/components/base_components/kubernetes_app.py

```diff
@@ -21,19 +28,92 @@ class Config(CamelCaseConfig):
 
 
 # TODO: label and annotations
-
-
 class KubernetesApp(PipelineComponent):
     """Base kubernetes app"""
 
     _type = "kubernetes-app"
     app: KubernetesAppConfig
 
+    version: str | None = None
+
+    class Config:
+        keep_untouched = (cached_property,)
+
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.check_compatible_name()
+        self.__check_compatible_name()
+
+    @cached_property
+    def helm_wrapper(self) -> Helm:
+        helm_wrapper = Helm(self.config.helm_config)
+        helm_repo_config = self.get_helm_repo_config()
+        if helm_repo_config is not None:
+            helm_wrapper.add_repo(
+                helm_repo_config.repository_name,
+                helm_repo_config.url,
+                helm_repo_config.repo_auth_flags,
+            )
+        return helm_wrapper
+
+    @cached_property
+    def helm_diff(self) -> HelmDiff:
+        return HelmDiff(self.config.helm_diff_config)
+
+    @property
+    def helm_release_name(self) -> str:
+        """The name for the Helm release. Can be overridden."""
+        return self.name
+
+    @property
+    def namespace(self) -> str:
+        return self.app.namespace
+
+    def deploy(self, dry_run: bool) -> None:
+        stdout = self.helm_wrapper.upgrade_install(
+            self.helm_release_name,
+            self.get_helm_chart(),
+            dry_run,
+            self.namespace,
+            self.to_helm_values(),
+            HelmUpgradeInstallFlags(version=self.get_helm_chart_version()),
+        )
+
+        if dry_run and self.helm_diff.config.enable:
+            self.print_helm_diff(stdout)
+
+    # TODO: Separate destroy and clean
+    def destroy(self, dry_run: bool, clean: bool, delete_outputs: bool) -> None:
+        stdout = self.helm_wrapper.uninstall(
+            self.namespace,
+            self.helm_release_name,
+            dry_run,
+        )
+        if dry_run and self.helm_diff.config.enable:
+            self.print_helm_diff(stdout)
+
+    def to_helm_values(self) -> dict:
+        return self.app.dict(by_alias=True, exclude_none=True, exclude_unset=True)
+
+    def print_helm_diff(self, stdout: str):
+        current_release = self.helm_wrapper.get_manifest(
+            self.helm_release_name, self.namespace
+        )
+        new_release = Helm.load_helm_manifest(stdout)
+        helm_diff = HelmDiff.get_diff(current_release, new_release)
+        self.helm_diff.log_helm_diff(helm_diff, log)
+
+    def get_helm_repo_config(self) -> HelmRepoConfig | None:
+        return None
+
+    def get_helm_chart(self) -> str:
+        raise NotImplementedError(
+            f"Please implement the get_helm_chart() method of the {self.__module__} module."
+        )
+
+    def get_helm_chart_version(self) -> str | None:
+        return self.version
```

Comment:
what is the purpose of this function? public attributes can be accessed directly, there's no need for a getter-function in Python. If it should be "read-only" make it a `@property`